platform/event/event_descriptor.proto

```diff
@@ -0,0 +1,64 @@
+/**
+ * # Event Descriptor
+ * Unique identifier for an event.
+ *
+ * ### Keywords
+ * The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+ * "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+ * document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119).
+ */
+syntax = "proto3";
+
+package com.hedera.hapi.platform.event;
+
+/*
+ * Hedera Network Services Protobuf
+ *
+ * Copyright (C) 2018 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+option java_package = "com.hedera.hapi.platform.event";
+// <<<pbj.java_package = "com.hedera.hapi.platform.event">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * Unique identifier for an event.
+ */
+message EventDescriptor {
+  /**
+   * The hash of the event. <br/>
+   * Future work: the precise hash methodology hasn't been determined yet.
+   */
+  bytes hash = 1;
```

Comment:
If we go with `bytes` type for hashes then that is minimal wasted bytes. We will need to specify that it is a 48 byte SHA384 hash in documentation. It means if we want to change it in the future we will need to add a new field with new number for that hash like: `bytes hash_sha512 = 10;` which will work fine. The other option is to have a Hash message with a enum for hash type. We already have one but unfortunately the enum is frustrating as the default value is UNKNOWN.  Default values are special as they require 0 bytes to be sent on wire. So if we use existing hash object we are adding 2 bytes for the message wrapper and 2 bytes for enum. So it costs us 4 extra bytes every time, a small amount of extra parsing serialization effort and 2 extra Java object creations.

Reply:
My personal preference is to use the raw `bytes` field, and explicitly state what kind of hash it is

@lpetrovic05 Any preference here?

Reply:
Previously we have decided to go with bytes for hashes and signatures, I think for performance reasons. This is how I did it for the state signature payload. So just mention that is should be a SHA384 hash.

Reply:
Specified the hash type